api/v1alpha1/companion_types.go

```diff
@@ -17,37 +17,105 @@ limitations under the License.
 package v1alpha1
 
 import (
+	kcorev1 "k8s.io/api/core/v1"
 	kmetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-
 const (
 	StateReady      string = "Ready"
 	StateError      string = "Error"
 	StateProcessing string = "Processing"
+	StateDeleting   string = "Deleting"
 	StateWarning    string = "Warning"
 )
 
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
 // CompanionSpec defines the desired state of Companion.
 type CompanionSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+
+	// Companion configuration.
+	// +kubebuilder:default:={namespace:"companion",deploymentNamespace:"ai-core",configMapNames:{"companion-config"},secretsNames:{"companion-secrets"},replicas:1}
+	CompanionConfig CompanionConfig `json:"companionConfig"`
+
+	// HANA Cloud configuration.
+	// +kubebuilder:default:={namespace:"hana-cloud"}
+	HanaCloudConfig HanaCloudConfig `json:"hanaCloudConfig"`
+
+	// Redis configuration.
+	// +kubebuilder:default:={namespace:"redis"}
+	RedisConfig RedisConfig `json:"redisConfig"`
+
+	// Container port for the companion backend. Default value is 5000.
+	// ContainerPort int32 `json:"containerPort"`
```

Comment:
Why is this commented out? Could we just delete it?

Reply:
Please focus on concept. And the structure. This is only an initial draft.